src/payment/strategies/sdk/strategy.ts

```diff
@@ -0,0 +1,155 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import { handler } from './handler';
+import { initializers, PaymentProcessor } from './initializers';
+
+/*
+    This strategy would be used for all PPSDK methods
+
+    Key points:
+        - The use of a "PPSDK Response Handler", which would
+            (1) discern different code paths based on success/failure/error/continue
+                responses from our checkout/processing endpoint (received via HostedFormFactory above)
+            (2) not return control to the strategy until all continue steps are exhausted
+            (3) attempt to handle any recoverable failure responses(?)
+            (4) translate error and un-recoverable failure responses into Checkout SDK errors
+                (e.g. those from src/common/error/errors/index.ts)
+                before throwing
+            (5) Delegate and await the resolving of relevant(?) continue steps to checkout-js
+            (6) Return a resolving promise from 'execute'
+                upon a success message from the checkout/processing endpoint
+
+*/
+
+export class Strategy implements PaymentStrategy {
+    protected _paymentProcessor?: PaymentProcessor;
+
+    constructor(
+        protected _store: CheckoutStore,
+        protected _orderActionCreator: OrderActionCreator
+    ) {}
+
+    async initialize(options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        if (!options || !options.methodId) {
+            // TODO: Can we make this object and attribute mandatory?
```

Comment:
this is mandatory but it's always provided automatically by the registry, so there's nothing to really about at this stage. If you are talking about not making it mandatory at the entry point (checkoutService.submitOrder), how would the SDK be aware of what payment strategy it should use?

Reply:
I was wanting to make it mandatory (right now the type has it as optional: https://github.com/bigcommerce/checkout-sdk-js/blob/67b1e0a818f16ba94b729683ea31362907980fad/src/payment/strategies/payment-strategy.ts#L10 )

But I guess if other consumers are already using it without passing in options, I might be stuck with my "not undefined" checks. 

I think I'll organise a meeting to clear up what this proposal needs from the Payment Method object, as the use of `gateway` vs `provider` vs `type` is all a little confusing.